sympy/sets/sets.py

```diff
@@ -652,6 +713,35 @@ def _boundary(self):
     def _measure(self):
         raise NotImplementedError("(%s)._measure" % self)
 
+    def _kind(self):
+        return SetKind(UndefinedKind)
+
+    def _helper_kind(self):
+        if not self.args:
+            return SetKind()
+        elif all(i.kind == self.args[0].kind for i in self.args):
+            return SetKind(self.args[0].kind)
+        else:
+            return SetKind(UndefinedKind)
+
+    def _union_setkind(self):
+        args = tuple(arg.kind for arg in self.args if arg is not S.EmptySet)
```

Comment:
These methods would be more generically useful if they didn't access `self.args`.

Reply:
Should I make them static function,  with argument self.args

Reply:
Better to use `classmethod`.

Reply:
But then these method would use cls.args inside the method and only argument of these method will be cls,  which would make it similar to self.args

Reply:
The point is that the methods should not presume anything about the args of self because self could be an instance of many possible different subclasses that prescribe different meanings to their args. The method as written here presumes that self is a union so it might as well be a method in the Union class rather than here.

Reply:
```
    721         if not cls.args:
    722             return SetKind()
--> 723         elif all(i.kind == cls.args[0].kind for i in cls.args):
    724             return SetKind(cls.args[0].kind)
    725         else:

TypeError: 'property' object is not iterable
```
Does it mean `args` is attached to self or instance and not class itself?

Reply:
The args will need to be passed into this method.

Reply:
Then it will become static method since cls or self is not needed inside method, only argument that needed to passed will be `self.args` when calling this method in different, Is this good, shall I implement this?

Reply:
Should I also `union_setkind` and `intersection_setkind` to their respective classes?